main.tf

```diff
@@ -166,6 +166,11 @@ module "container_definition" {
       hostPort      = 0
       protocol      = "tcp"
     },
+    var.enable_prometheus_metrics ? {
+      containerPort = 9100
+      hostPort      = 0
+      protocol      = "tcp"
+    } : null,
```

Comment:
if you do it like this, than you would need to wrap the whole list with compact (see: https://developer.hashicorp.com/terraform/language/functions/compact) to ensure you don't have null values in your list.

However i prefer a different way of writing it:
```terraform
port_mappings = concat([
    ... (always existing entries)
    ], var.enable_prometheus_metrics ? [your-entry] : []
)
```

Reply:
Since `port_mappings` is list of object I'm unable to use compact function for it, do you have another recommendation for that.

Reply:
in fact compact would ony have been necessary if you wanted to do it your way. Since you are following my other suggestion (which is also my preference), you don't need compact at all.

Reply:
true, thanks for the guidance.